pkg/controller/servicebindingrequest/planner.go

```diff
@@ -0,0 +1,177 @@
+package servicebindingrequest
+
+import (
+	"context"
+	"strings"
+
+	"github.com/go-logr/logr"
+	olmv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ustrv1 "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	v1alpha1 "github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1"
+)
+
+// Planner plans resources needed to bind a given backend service, using OperatorLifecycleManager
+// standards and CustomResourceDefinitionDescription data to understand which attributes are needed.
+type Planner struct {
+	ctx    context.Context                 // request context
+	client dynamic.Interface               // kubernetes dynamic api client
+	sbr    *v1alpha1.ServiceBindingRequest // instantiated service binding request
+	logger logr.Logger                     // logger instance
+}
+
+// Plan outcome, after executing planner.
+type Plan struct {
+	Ns             string                      // namespace name
+	Name           string                      // plan name, same than ServiceBindingRequest
+	CRDDescription *olmv1alpha1.CRDDescription // custom resource definition description
+	CR             *ustrv1.Unstructured        // custom resource object
+}
+
+// searchCRDDescription based on BackingServiceSelector instance, find a
+// CustomResourceDefinitionDescription to return, otherwise creating a not-found error.
+func (p *Planner) searchCRDDescription() (*olmv1alpha1.CRDDescription, error) {
+	csvResourceName := "clusterserviceversions"
+	ns := p.sbr.GetNamespace()
+	gvr := schema.GroupVersionResource{
+		Group:    olmv1alpha1.GroupName,
+		Version:  olmv1alpha1.GroupVersion,
+		Resource: csvResourceName,
+	}
+
+	logger := p.logger.WithValues(
+		"CSV.Group", gvr.Group,
+		"CSV.Version", gvr.Version,
+		"CSV.Resource", gvr.Resource,
+		"CSV.Namespace", ns,
+	)
+	logger.Info("Searching for ClusterServiceVersion...")
+
+	// FIXME: usually the CSV resources are in the "openshift-operator-lifecycle-manager" namespace;
```

Comment:
Umm. no. The CSVs which a user wants to use would be copied to the user's namespace. Most CSVs get copied over to all namespaces.